src/main/java/com/juniorstart/juniorstart/service/UserProfileAvatarService.java

```diff
@@ -0,0 +1,103 @@
+package com.juniorstart.juniorstart.service;
+
+import com.juniorstart.juniorstart.exception.BadRequestException;
+import com.juniorstart.juniorstart.exception.ResourceNotFoundException;
+import com.juniorstart.juniorstart.model.UserProfile;
+import com.juniorstart.juniorstart.repository.UserProfileRepository;
+import com.juniorstart.juniorstart.security.UserPrincipal;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+import org.springframework.web.multipart.MultipartFile;
+
+import java.io.IOException;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.zip.DataFormatException;
+import java.util.zip.Deflater;
+import java.util.zip.Inflater;
+
+import java.io.ByteArrayOutputStream;
+
+
+@Service
+public class UserProfileAvatarService {
+
+    final private UserProfileRepository userProfileRepository;
+
+    UserProfileAvatarService(UserProfileRepository userProfileRepository) {
+        this.userProfileRepository = userProfileRepository;
+    }
+
+    public ResponseEntity<?> getUserAvatar(UserPrincipal currentUser) {
+
+        Optional<UserProfile> foundUser = findUser(currentUser.getId());
+
+        return ResponseEntity.ok(decompressBytes(foundUser.get().getUserAvatar()));
+    }
+
+    public ResponseEntity<?> addUserAvatar(MultipartFile file, UserPrincipal currentUser) throws IOException {
+
+        Optional<UserProfile> foundUser = findUser(currentUser.getId());
+
+        foundUser.get().setUserAvatar(compressBytes(file.getBytes()));
+
+        return ResponseEntity.ok(userProfileRepository.save(foundUser.get()));
```

Comment:
Nie lepiej zwracać ApiResponse że po prostu akcja aktualizacji, dodania usunięcia itp się udała ?

Reply:
Jest zwracana 200, Dla Frontu to powinno wystarczyć.

Reply:
ApiResponse przyjeliśmy już w projekcie

Reply:
@hosu794  a po co jest zastosowane te ApiResponse  ?